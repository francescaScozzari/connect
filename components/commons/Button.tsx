import styled from 'styled-components'

export const Button = styled.button`
  align-items: center;
  background-color: white;
  border: 2px solid black;
  border-radius: 10px;
  cursor: pointer;
  display: flex;
  justify-content: center;
  min-width: 120px;
  outline: none;
  padding: 1em 1.5em;
  gap: 1em;
  transition: 0.3s background-color ease-in-out;
  width: fit-content;

  &:disabled {
    cursor: not-allowed;
    opacity: 0.4;
  }
`
