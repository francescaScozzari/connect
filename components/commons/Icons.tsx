import React from 'react'

type Props = {
  title: string
} & React.ComponentPropsWithoutRef<'svg'>

const IconArrowCircle = ({ title, ...props }: Props) => {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" {...props}>
      <title>{title}</title>
      <path d="M12 1a11 11 0 1 0 11 11A11 11 0 0 0 12 1Zm4.707 9.707a1 1 0 0 1-1.414 0L13 8.414V18a1 1 0 0 1-2 0V8.414l-2.293 2.293a1 1 0 1 1-1.414-1.414l4-4a1 1 0 0 1 1.414 0l4 4a1 1 0 0 1 0 1.414Z" />
    </svg>
  )
}

const IconArrowSubmit = ({ title, ...props }: Props) => {
  return (
    <svg
      width="45"
      height="45"
      viewBox="0 0 45 45"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <title>{title}</title>
      <g id="Arrow forward">
        <path
          id="Vector"
          d="M22.5 7.5L19.8563 10.1437L30.3188 20.625H7.5V24.375H30.3188L19.8563 34.8563L22.5 37.5L37.5 22.5L22.5 7.5Z"
          fill="black"
        />
      </g>
    </svg>
  )
}

const IconCrossReset = ({ title, ...props }: Props) => {
  return (
    <svg
      width="31"
      height="31"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <title>{title}</title>
      <g>
        <path
          d="m24.5417 8.27962-1.8212-1.82125L15.5 13.6788 8.27962 6.45837 6.45837 8.27962 13.6788 15.5l-7.22043 7.2205 1.82125 1.8212L15.5 17.3213l7.2205 7.2204 1.8212-1.8212L17.3213 15.5l7.2204-7.22038Z"
          fill="#000"
        />
      </g>
    </svg>
  )
}

const IconCopy = ({ title, ...props }: Props) => {
  return (
    <svg
      width="49"
      height="49"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <title>{title}</title>
      <g>
        <path
          d="M32.6673 2.04163H8.16732c-2.24584 0-4.08334 1.8375-4.08334 4.08333V34.7083h4.08334V6.12496H32.6673V2.04163Zm-2.0416 8.16667H16.334c-2.2458 0-4.0629 1.8375-4.0629 4.0833l-.0204 28.5834c0 2.2458 1.817 4.0833 4.0629 4.0833h22.4787c2.2459 0 4.0834-1.8375 4.0834-4.0833V22.4583l-12.25-12.25ZM16.334 42.875V14.2916h12.25V24.5h10.2083v18.375H16.334Z"
          fill="#000"
        />
      </g>
    </svg>
  )
}

const IconTarget = ({ title, ...props }: Props) => {
  return (
    <svg
      width="49"
      height="51"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <title>{title}</title>
      <path
        d="M24.4993 16.9969c-4.512 0-8.1666 3.803-8.1666 8.4984s3.6546 8.4985 8.1666 8.4985c4.5121 0 8.1667-3.8031 8.1667-8.4985 0-4.6954-3.6546-8.4984-8.1667-8.4984Zm18.2525 6.3738c-.9391-8.8596-7.697-15.89205-16.2108-16.86937V4.24924c0-1.16853-.9187-2.12461-2.0417-2.12461-1.1229 0-2.0416.95608-2.0416 2.12461v2.25209C13.9439 7.47865 7.18602 14.5111 6.24685 23.3707H4.08268c-1.12291 0-2.04166.9561-2.04166 2.1246 0 1.1686.91875 2.1246 2.04166 2.1246h2.16417c.93917 8.8597 7.69705 15.8921 16.21085 16.8694v2.2521c0 1.1686.9187 2.1246 2.0416 2.1246 1.123 0 2.0417-.956 2.0417-2.1246v-2.2521c8.5138-.9773 15.2717-8.0097 16.2108-16.8694h2.1642c1.1229 0 2.0417-.956 2.0417-2.1246 0-1.1685-.9188-2.1246-2.0417-2.1246h-2.1642ZM24.4993 40.3676c-7.9012 0-14.2916-6.65-14.2916-14.8723 0-8.2222 6.3904-14.8722 14.2916-14.8722 7.9013 0 14.2917 6.65 14.2917 14.8722 0 8.2223-6.3904 14.8723-14.2917 14.8723Z"
        fill="#000"
      />
    </svg>
  )
}

export {
  IconArrowCircle,
  IconArrowSubmit,
  IconCrossReset,
  IconCopy,
  IconTarget
}
