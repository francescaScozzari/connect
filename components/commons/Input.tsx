import React from 'react'
import styled from 'styled-components'

type Props = JSX.IntrinsicElements['input'] & {
  errors?: string[]
}

const InputWithErrors = ({
  errors,
  max,
  min,
  onChange,
  placeholder,
  required,
  step,
  style,
  type,
  value
}: Props) => {
  return (
    <InputContainer style={style}>
      <Input
        errors={errors && errors.length > 0}
        max={max}
        min={min}
        onChange={onChange}
        placeholder={placeholder}
        required={required}
        step={step}
        type={type}
        value={value}
      />
      {errors && errors.length > 0 && errors.map(err => <p key={err}>{err}</p>)}
    </InputContainer>
  )
}

const Input = styled.input<{ errors?: boolean }>`
  border: 2px solid black;
  box-sizing: border-box;
  border-radius: 0.5em;
  padding: 0.625em 2em;
  width: 25em;
  font-style: normal;
  font-size: 1.125rem;
  &::placeholder {
    padding-left: 1.5em;
    font-size: 1.125rem;
    background: url("data:image/svg+xml,%3Csvg height='24' width='24' xmlns='http://www.w3.org/2000/svg'%3E%3Cdefs%3E%3Cstyle%3E.cls-1%7Bfill:none;stroke:%23000;stroke-linecap:round;stroke-linejoin:round;stroke-width:2px%7D%3C/style%3E%3C/defs%3E%3Cg id='_21.search'%3E%3Ccircle class='cls-1' cx='9' cy='9' r='8'/%3E%3Cpath class='cls-1' d='m15 15 8 8'/%3E%3C/g%3E%3C/svg%3E")
      no-repeat 0.5em center;
  }
  @media (max-width: 768px) {
    font-size: 1em;
    width: 15em;
  }
`

const InputContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 1.5em;
  p {
    font-size: 0.75em;
    margin-left: 0.625em;
    color: ${({ theme }) => theme.colors.status.error};
  }
`

export { Input, InputWithErrors }
