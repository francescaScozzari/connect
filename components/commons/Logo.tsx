import React from 'react'

type Props = {
  title: string
} & React.ComponentPropsWithoutRef<'svg'>

const BigLogo = ({ title, ...props }: Props) => {
  return (
    <svg
      width="395"
      height="122"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <title>{title}</title>
      <path
        d="M56.8333 5.38354H11.7031V52.7563c5.5773-25.1345 20.4454-40.9681 45.1302-47.37276ZM11.6992 70.4567v46.7523h46.126c-25.255-6.107-40.4903-20.8972-46.126-46.7523ZM67.5977 105.379c-2.1168-11.6491-6.8235-21.3173-14.7819-29.202-7.9306-7.8598-17.5191-12.6285-29.0853-14.5482 24.2424-5.2918 38.7294-19.6981 44.0619-44.1704 1.9027 11.8438 6.7846 21.4147 14.718 29.3579 7.9333 7.9432 17.5946 12.609 29.0996 14.6149-24.1538 5.1499-38.5908 19.431-44.0123 43.9478ZM123.505 69.9977v47.2113H77.4453c25.2547-6.107 40.4207-21.3562 46.0597-47.2113ZM78.3652 5.38354h45.1388V52.5699c-5.58-25.1345-20.457-40.7817-45.1388-47.18636ZM145.585 115.161c0 1.517-1.007 2.382-3.021 2.382h-3.652v-9.006h3.419c1.755 0 2.823.809 2.823 2.304 0 1.068-.551 1.763-1.371 2.058 1.16.242 1.802 1.016 1.802 2.265v-.003Zm-6.075-6.059v3.586h2.796c1.416 0 2.236-.604 2.236-1.8 0-1.197-.82-1.786-2.236-1.786h-2.796Zm5.477 6.006c0-1.285-.854-1.852-2.423-1.852h-3.054v3.717h3.054c1.569 0 2.423-.568 2.423-1.865ZM147.076 113.038c0-2.612 1.814-4.568 4.262-4.568 2.448 0 4.261 1.942 4.261 4.568 0 2.627-1.827 4.566-4.261 4.566s-4.262-1.956-4.262-4.566Zm7.914 0c0-2.278-1.569-3.964-3.652-3.964-2.084 0-3.664 1.686-3.664 3.964 0 2.279 1.569 3.962 3.664 3.962 2.094 0 3.652-1.683 3.652-3.962ZM157.094 113.038c0-2.612 1.813-4.568 4.261-4.568 2.448 0 4.262 1.942 4.262 4.568 0 2.627-1.828 4.566-4.262 4.566-2.434 0-4.261-1.956-4.261-4.566Zm7.914 0c0-2.278-1.569-3.964-3.653-3.964-2.083 0-3.663 1.686-3.663 3.964 0 2.279 1.569 3.962 3.663 3.962 2.095 0 3.653-1.683 3.653-3.962ZM166.949 116.383l.27-.487c.562.643 1.605 1.129 2.704 1.129 1.641 0 2.364-.784 2.364-1.761 0-2.701-5.115-1.082-5.115-4.398 0-1.272.89-2.393 2.901-2.393.901 0 1.839.309 2.47.81l-.222.528c-.679-.512-1.511-.759-2.248-.759-1.591 0-2.306.809-2.306 1.8 0 2.701 5.116 1.107 5.116 4.373 0 1.275-.924 2.382-2.949 2.382-1.218 0-2.387-.501-2.985-1.221v-.003ZM176.818 109.127h-2.949v-.593h6.498v.593h-2.951v8.413h-.595v-8.413h-.003ZM181.955 108.534h.598v9.006h-.598v-9.006ZM192.031 108.534v9.006h-.493l-5.583-7.824v7.824h-.598v-9.006h.504l5.572 7.824v-7.824h.598ZM200.83 113.075h.575v3.369c-.737.771-1.802 1.157-2.926 1.157-2.47 0-4.284-1.942-4.284-4.565 0-2.624 1.814-4.569 4.298-4.569 1.182 0 2.223.412 2.949 1.236l-.376.436c-.715-.77-1.569-1.068-2.565-1.068-2.13 0-3.71 1.7-3.71 3.965 0 2.264 1.58 3.962 3.71 3.962.865 0 1.661-.231 2.329-.821v-3.102ZM206.93 108.534h.598v9.006h-.598v-9.006ZM217.005 108.534v9.006h-.492l-5.583-7.824v7.824h-.598v-9.006h.504l5.571 7.824v-7.824h.598ZM225.806 113.075h.576v3.369c-.737.771-1.803 1.157-2.926 1.157-2.47 0-4.284-1.942-4.284-4.565 0-2.624 1.814-4.569 4.298-4.569 1.182 0 2.222.412 2.948 1.236l-.375.436c-.715-.77-1.569-1.068-2.565-1.068-2.131 0-3.711 1.7-3.711 3.965 0 2.264 1.58 3.962 3.711 3.962.865 0 1.661-.231 2.328-.821v-3.102ZM234.628 116.947v.593h-5.655v-9.006h5.48v.593h-4.882v3.55h4.367v.578h-4.367v3.692h5.057ZM243.273 108.534v9.006h-.492l-5.583-7.824v7.824h-.598v-9.006h.503l5.572 7.824v-7.824h.598ZM246.09 108.534h.598v9.006h-.598v-9.006ZM249.442 113.717v-5.183h.596v5.158c0 2.251.984 3.306 2.692 3.306 1.708 0 2.682-1.055 2.682-3.306v-5.158h.595v5.183c0 2.574-1.252 3.884-3.291 3.884-2.039 0-3.277-1.31-3.277-3.884h.003ZM266.518 117.54l-.011-7.707-3.429 6.511h-.293l-3.429-6.486v7.682h-.585v-9.006h.504l3.663 6.958 3.664-6.958h.492l.014 9.006h-.59ZM273.459 109.127v3.845h4.368v.592h-4.368v3.976h-.598v-9.006h5.48v.593h-4.882ZM279.504 113.038c0-2.612 1.814-4.568 4.261-4.568 2.448 0 4.262 1.942 4.262 4.568 0 2.627-1.828 4.566-4.262 4.566-2.434 0-4.261-1.956-4.261-4.566Zm7.914 0c0-2.278-1.569-3.964-3.653-3.964-2.083 0-3.663 1.686-3.663 3.964 0 2.279 1.569 3.962 3.663 3.962 2.095 0 3.653-1.683 3.653-3.962ZM295.882 117.54l-1.956-3.01c-.258.039-.514.064-.795.064h-2.34v2.946h-.598v-9.006h2.938c2.061 0 3.302 1.132 3.302 3.035 0 1.428-.704 2.418-1.92 2.819l2.048 3.152h-.679Zm-2.751-3.525c1.755 0 2.692-.901 2.692-2.443 0-1.541-.937-2.445-2.692-2.445h-2.34v4.888h2.34ZM307.179 116.947v.593h-5.656v-9.006h5.48v.593h-4.881v3.55h4.367v.578h-4.367v3.692h5.057ZM314.841 117.54l-2.821-4.168-2.809 4.168h-.701l3.149-4.643-2.927-4.363h.701l2.612 3.873 2.61-3.873h.667l-2.926 4.349 3.149 4.657h-.704ZM316.264 113.038c0-2.626 1.813-4.568 4.25-4.568 1.157 0 2.2.412 2.915 1.249l-.375.423c-.701-.759-1.558-1.068-2.518-1.068-2.097 0-3.677 1.7-3.677 3.964 0 2.265 1.58 3.962 3.677 3.962.96 0 1.814-.32 2.518-1.079l.375.426c-.715.834-1.755 1.26-2.915 1.26-2.434 0-4.25-1.942-4.25-4.566v-.003ZM331.052 116.947v.593h-5.656v-9.006h5.48v.593h-4.881v3.55h4.367v.578h-4.367v3.692h5.057ZM333.027 108.534h.598v8.413h4.718v.593h-5.316v-9.006ZM339.967 108.534h.598v8.413h4.718v.593h-5.316v-9.006ZM352.563 116.947v.593h-5.655v-9.006h5.48v.593h-4.882v3.55h4.368v.578h-4.368v3.692h5.057ZM361.21 108.534v9.006h-.492l-5.583-7.824v7.824h-.598v-9.006h.504l5.571 7.824v-7.824h.598ZM363.373 113.038c0-2.626 1.814-4.568 4.25-4.568 1.158 0 2.201.412 2.916 1.249l-.376.423c-.701-.759-1.558-1.068-2.517-1.068-2.098 0-3.678 1.7-3.678 3.964 0 2.265 1.58 3.962 3.678 3.962.959 0 1.813-.32 2.517-1.079l.376.426c-.715.834-1.756 1.26-2.916 1.26-2.434 0-4.25-1.942-4.25-4.566v-.003ZM378.159 116.947v.593h-5.655v-9.006h5.48v.593h-4.882v3.55h4.367v.578h-4.367v3.692h5.057ZM193.275 49.287c7.992 4.165 13.008 12.5032 13.008 22.6778 0 15.1046-11.859 26.5451-27.514 26.5451h-39.091V5.29187h38.309c15.038 0 26.012 10.08553 26.012 23.85743 0 8.33-4.245 15.8225-10.724 20.1377ZM181.82 33.4033c0-4.8382-3.015-8.1157-7.902-8.1157h-11.809v16.2315h11.809c4.887 0 7.902-3.2775 7.902-8.1158Zm1.349 35.5094c0-4.9495-3.307-8.4384-8.367-8.4384h-12.693v16.952h12.693c5.06 0 8.367-3.4861 8.367-8.5164v.0028ZM217.033 5.28894h23.675V98.507h-23.675V5.28894ZM313.862 82.5982h-7.279v15.9115h-20.704V82.5982h-38.68v-4.3319l25.812-68.3172h21.396l-19.956 52.8036h12.885l4.726-19.5033h14.518v19.5033h7.28v19.8427l.002.0028ZM378.16 76.2326V98.507h-55.295V5.28894h54.268V27.4882h-31.004v13.7219h27.855v21.0641h-27.855v13.9556h32.031v.0028Z"
        fill="#000"
      />
    </svg>
  )
}

const SmallLogo = ({ title, ...props }: Props) => {
  return (
    <svg
      width="202"
      height="61"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <title>{title}</title>
      <path
        d="M29.064 2.692H5.984v23.686C8.838 13.811 16.44 5.894 29.065 2.692Z"
        fill="#009ADE"
      />
      <path
        d="M5.983 35.228v23.377h23.589C16.657 55.55 8.865 48.154 5.983 35.228Z"
        fill="#ABC511"
      />
      <path
        d="M34.57 52.69c-1.083-5.825-3.49-10.66-7.56-14.601-4.056-3.93-8.96-6.315-14.874-7.275C24.533 28.17 31.942 20.965 34.669 8.73c.973 5.922 3.47 10.708 7.526 14.68 4.057 3.97 8.998 6.304 14.882 7.307C44.725 33.29 37.342 40.43 34.569 52.69Z"
        fill="#fff"
      />
      <path
        d="M63.16 34.999v23.605H39.604C52.52 55.552 60.275 47.927 63.159 35Z"
        fill="#DF057E"
      />
      <path
        d="M40.076 2.692h23.083v23.593C60.306 13.718 52.698 5.895 40.076 2.692Z"
        fill="#0C927A"
      />
      <path
        d="M74.451 57.58c0 .759-.515 1.192-1.545 1.192H71.04v-4.503h1.748c.897 0 1.444.404 1.444 1.151 0 .535-.282.882-.702 1.03.594.12.922.508.922 1.132v-.001Zm-3.107-3.03v1.794h1.43c.724 0 1.144-.302 1.144-.9 0-.598-.42-.893-1.144-.893h-1.43Zm2.801 3.004c0-.642-.436-.926-1.239-.926h-1.562v1.858h1.562c.803 0 1.24-.283 1.24-.932ZM75.214 56.52c0-1.307.927-2.285 2.18-2.285 1.251 0 2.178.971 2.178 2.284 0 1.314-.934 2.283-2.179 2.283-1.245 0-2.18-.978-2.18-2.283Zm4.047 0c0-1.14-.802-1.983-1.868-1.983-1.065 0-1.873.843-1.873 1.982 0 1.14.802 1.981 1.873 1.981s1.868-.841 1.868-1.98ZM80.336 56.52c0-1.307.928-2.285 2.18-2.285 1.252 0 2.18.971 2.18 2.284 0 1.314-.936 2.283-2.18 2.283-1.245 0-2.18-.978-2.18-2.283Zm4.047 0c0-1.14-.802-1.983-1.867-1.983-1.066 0-1.874.843-1.874 1.982 0 1.14.803 1.981 1.874 1.981 1.07 0 1.867-.841 1.867-1.98ZM85.376 58.191l.138-.243c.288.321.821.565 1.383.565.84 0 1.21-.392 1.21-.88 0-1.352-2.617-.542-2.617-2.2 0-.636.456-1.196 1.484-1.196.46 0 .94.154 1.263.404l-.114.265a1.98 1.98 0 0 0-1.149-.38c-.814 0-1.18.405-1.18.9 0 1.35 2.617.554 2.617 2.187 0 .637-.472 1.19-1.508 1.19-.623 0-1.22-.25-1.527-.61v-.002ZM90.423 54.563h-1.508v-.296h3.324v.296h-1.51v4.207h-.304v-4.207h-.002ZM93.05 54.267h.307v4.503h-.306v-4.503ZM98.203 54.267v4.503h-.252l-2.855-3.912v3.912h-.306v-4.503h.258l2.85 3.912v-3.912h.305ZM102.703 56.537h.294v1.685c-.377.385-.922.579-1.496.579-1.263 0-2.19-.971-2.19-2.283 0-1.312.927-2.284 2.197-2.284.604 0 1.136.206 1.508.617l-.192.219c-.366-.386-.803-.535-1.312-.535-1.09 0-1.897.85-1.897 1.983 0 1.132.807 1.98 1.897 1.98.443 0 .849-.115 1.191-.41v-1.55ZM105.822 54.267h.306v4.503h-.306v-4.503ZM110.975 54.267v4.503h-.252l-2.855-3.912v3.912h-.306v-4.503h.257l2.85 3.912v-3.912h.306ZM115.476 56.537h.294v1.685c-.377.385-.922.579-1.496.579-1.263 0-2.191-.971-2.191-2.283 0-1.312.928-2.284 2.198-2.284.604 0 1.136.206 1.508.617l-.192.219c-.366-.386-.803-.535-1.312-.535-1.09 0-1.898.85-1.898 1.983 0 1.132.808 1.98 1.898 1.98.443 0 .849-.115 1.191-.41v-1.55ZM119.987 58.474v.296h-2.892v-4.503h2.802v.296h-2.496v1.775h2.233v.29h-2.233v1.846h2.586ZM124.408 54.267v4.503h-.252l-2.855-3.912v3.912h-.306v-4.503h.258l2.849 3.912v-3.912h.306ZM125.849 54.267h.305v4.503h-.305v-4.503ZM127.563 56.859v-2.592h.304v2.58c0 1.125.504 1.652 1.377 1.652.874 0 1.372-.527 1.372-1.653v-2.579h.304v2.592c0 1.286-.64 1.942-1.683 1.942-1.042 0-1.675-.656-1.675-1.942h.001ZM136.296 58.77l-.006-3.853-1.754 3.255h-.149l-1.754-3.243v3.841h-.299v-4.503h.257l1.874 3.48 1.873-3.48h.252l.007 4.503h-.301ZM139.845 54.563v1.923h2.234v.296h-2.234v1.988h-.305v-4.503h2.802v.296h-2.497ZM142.937 56.52c0-1.307.927-2.285 2.179-2.285 1.252 0 2.179.971 2.179 2.284 0 1.314-.934 2.283-2.179 2.283-1.245 0-2.179-.978-2.179-2.283Zm4.047 0c0-1.14-.803-1.983-1.868-1.983-1.066 0-1.874.843-1.874 1.982 0 1.14.803 1.981 1.874 1.981 1.071 0 1.868-.841 1.868-1.98ZM151.312 58.77l-1-1.505c-.132.02-.263.032-.406.032h-1.197v1.473h-.306v-4.503h1.503c1.054 0 1.688.566 1.688 1.518 0 .713-.36 1.209-.981 1.409l1.047 1.576h-.348Zm-1.406-1.763c.897 0 1.377-.45 1.377-1.22 0-.771-.48-1.224-1.377-1.224h-1.197v2.444h1.197ZM157.089 58.474v.296h-2.892v-4.503H157v.296h-2.497v1.775h2.234v.29h-2.234v1.846h2.586ZM161.007 58.77l-1.443-2.084-1.436 2.084h-.359l1.61-2.322-1.496-2.18h.358l1.336 1.935 1.334-1.936h.342l-1.497 2.174 1.611 2.33h-.36ZM161.735 56.52c0-1.314.928-2.285 2.174-2.285.592 0 1.125.206 1.491.625l-.192.211c-.359-.38-.797-.534-1.288-.534-1.072 0-1.88.85-1.88 1.982 0 1.133.808 1.981 1.88 1.981.491 0 .928-.16 1.288-.54l.192.213c-.366.418-.898.63-1.491.63-1.245 0-2.174-.97-2.174-2.282v-.002ZM169.297 58.474v.296h-2.892v-4.503h2.803v.296h-2.497v1.775h2.233v.29h-2.233v1.846h2.586ZM170.307 54.267h.306v4.207h2.413v.296h-2.719v-4.503ZM173.856 54.267h.306v4.207h2.413v.296h-2.719v-4.503ZM180.298 58.474v.296h-2.892v-4.503h2.802v.296h-2.496v1.775h2.233v.29h-2.233v1.846h2.586ZM184.72 54.267v4.503h-.252l-2.855-3.912v3.912h-.305v-4.503h.257l2.849 3.912v-3.912h.306ZM185.826 56.52c0-1.314.927-2.285 2.173-2.285.592 0 1.126.206 1.491.625l-.192.211c-.358-.38-.797-.534-1.287-.534-1.073 0-1.881.85-1.881 1.982 0 1.133.808 1.981 1.881 1.981.49 0 .927-.16 1.287-.54l.192.213c-.365.418-.897.63-1.491.63-1.244 0-2.173-.97-2.173-2.282v-.002ZM193.388 58.474v.296h-2.892v-4.503h2.802v.296h-2.496v1.775h2.233v.29h-2.233v1.846h2.586ZM98.84 24.643c4.086 2.083 6.651 6.252 6.651 11.34 0 7.552-6.064 13.272-14.07 13.272H71.43V2.645h19.59c7.691 0 13.303 5.043 13.303 11.93 0 4.164-2.171 7.91-5.484 10.068Zm-5.859-7.942c0-2.419-1.542-4.057-4.041-4.057H82.9v8.115h6.039c2.5 0 4.041-1.638 4.041-4.058Zm.69 17.755c0-2.475-1.691-4.22-4.279-4.22h-6.491v8.477h6.491c2.588 0 4.279-1.743 4.279-4.258v.001ZM110.989 2.644h12.107v46.61h-12.107V2.644ZM160.506 41.3h-3.723v7.955h-10.588v-7.956h-19.78v-2.166l13.2-34.158h10.942l-10.205 26.401h6.589l2.417-9.751h7.424v9.751h3.723v9.922l.001.001ZM193.388 38.116v11.138h-28.277V2.644h27.752v11.1h-15.855v6.861h14.245v10.532h-14.245v6.978h16.38v.001Z"
        fill="#fff"
      />
    </svg>
  )
}

export { BigLogo, SmallLogo }
