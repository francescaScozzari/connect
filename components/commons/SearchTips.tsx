import React from 'react'
import { styled } from 'styled-components'

import { Text } from '@/components/commons/Typography'
import { IconNote, IconTarget } from './Icons'

type Props = {
  titleColor?: string
  textColor?: string
}

const SearchTips = ({
  titleColor = '#2B2D42',
  textColor = '#242424'
}: Props) => {
  return (
    <Wrapper>
      <SubWrapper>
        <IconNote title={'note'} color={titleColor} />
        <Paragraph color={textColor}>
          <Title color={titleColor}>Start from the research call</Title>
          <br /> If you are working or will have to work on a research paper for
          a call, you might report in the search field the title or objectives
          of that
        </Paragraph>
      </SubWrapper>

      <SubWrapper>
        <IconTarget title={'target'} color={titleColor} />
        <Paragraph color={textColor}>
          <Title color={titleColor}>More detail for greater match</Title>
          <br /> Enter as much information, don't limit yourself to a single
          word, that way we can narrow it down and suggest the researchers best
          suited to your need
        </Paragraph>
      </SubWrapper>
    </Wrapper>
  )
}

const Wrapper = styled.div`
  display: flex;
  gap: 2em;
  max-width: 67em;
  margin: 4em 0;

  @media (max-width: 1280px) {
    max-width: 46em;
  }

  @media (max-width: 768px) {
    max-width: 37.5em;
  }
`

const SubWrapper = styled.div`
  display: flex;
  gap: 0.5em;
  padding: 1.5em;
  align-items: start;
  max-width: 29.85em;
  border: 1px solid #7a83ab;
  border-radius: 1.25em;

  svg {
    flex-shrink: 0;
  }
`

const Paragraph = styled(Text.Normal)<{ color: string }>`
  line-height: 1.5;
  color: ${({ color }) => color};
`

const Title = styled.span<{ color: string }>`
  display: inline-block;
  color: ${({ color }) => color};
  margin-bottom: 0.5em;
  font-weight: 600;
  line-height: 1.5;
`

export { SearchTips }
