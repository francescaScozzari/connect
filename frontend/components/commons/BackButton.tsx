import React from 'react'
import styled from 'styled-components'

import { IconSearch } from './Icons'
import { Text } from './Typography'

export const BackButton = () => {
  return (
    <Button>
      <IconSearch title="back" />
      <Label as="span">make a new search</Label>
    </Button>
  )
}

const Label = styled(Text.Normal)`
  text-transform: uppercase;
  font-weight: 600;
  line-height: normal;
  color: white;
`

const Button = styled.button`
  all: unset;
  display: flex;
  justify-content: center;
  align-content: center;
  border-radius: 10px;
  cursor: pointer;
  width: fit-content;
  outline: none;
  padding: 0.75em 1em;
  gap: 1em;
  transition: 0.3s background-color ease-in-out;
  background-color: ${({ theme }) => theme.colors.secondary[0]};

  &:disabled {
    cursor: not-allowed;
    opacity: 0.4;
  }
`
