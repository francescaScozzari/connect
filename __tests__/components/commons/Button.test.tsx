import { expect } from '@jest/globals'
import React from 'react'

import { Button } from '@/components/commons/Button'
import { renderWithWrappers } from '@/__tests__/functions'

const setup = () =>
  renderWithWrappers(<Button />)

describe('Button components', () => {
  describe('<Button />', () => {
    test('Button primary ui renders correctly', () => {
      const container = setup()
      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
