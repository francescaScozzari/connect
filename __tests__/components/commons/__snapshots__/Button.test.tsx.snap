// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button components <Button /> Button primary ui renders correctly 1`] = `
.c0 {
  align-items: center;
  background-color: #316CF4;
  border: 0;
  border-radius: 5px;
  color: #F1F9FF;
  cursor: pointer;
  display: flex;
  justify-content: center;
  min-width: 120px;
  outline: none;
  padding: 10px;
  transition: 0.3s background-color ease-in-out;
  width: fit-content;
}

.c0:disabled {
  cursor: not-allowed;
  opacity: 0.4;
}

.c0:hover {
  background-color: #007AFF;
}

<button
  class="c0"
/>
`;
